### PR TITLE
Add DataTypeA, DataTypeB (SHHS, HSHS)

### PR DESCRIPTION
**Pull request by KKyang**

### CODE REVIEW

File: tensilelite/Tensile/SolutionStructs.py [hunk 1]
```diff
@@ -451,7 +461,10 @@ def __str__(self):
 
     # DataTypes
     name += "_"
-    name += self["DataType"].toChar() # Type of A/B
+    if self["DataType"] == self["DataTypeA"] and self["DataType"] == self["DataTypeB"]:
+      name += self["DataType"].toChar() # Type of A/B
+    else:
+      name += self["DataTypeA"].toChar() + self["DataTypeB"].toChar()
```
[the remark this answers is not in the document]
Comment (KKyang) [reply tensilelite/Tensile/SolutionStructs.py:469]: Fixed

File: tensilelite/Tensile/Common.py [hunk 2]
```diff
@@ -1101,6 +1101,8 @@ def getArchitectureName(gfxName):
     "OperationType":            "GEMM",           # GEMM, TensorContraction, ConvolutionForward, ConvolutionBackwardData, ConvolutionBackwardWeights
 
     "DataType":                 0,                # data types can specified by a variety of ways, such as "s", as listed in SolutionStructs.py::DataType
+    "DataTypeA":                0,
```
Comment (jichangjichang) [inline tensilelite/Tensile/Common.py:1104]: describe the purpose here

File: tensilelite/Tensile/Source/client/include/TypedId.hpp [hunk 3]
```diff
@@ -115,5 +118,9 @@ namespace Tensile
     using TypedGemm_B8F8_S_S  = TypedGemm<BFloat8, Float8, float, float>;
     using TypedGemm_F8B8_B8_S = TypedGemm<Float8, BFloat8, BFloat8, BFloat8, float, float>;
     using TypedGemm_B8F8_B8_S = TypedGemm<BFloat8, Float8, BFloat8, BFloat8, float, float>;
+#ifdef TENSILE_USE_HALF
+    using TypedGemm_HF8_F8_S = TypedGemm<Half, Float8, Float8, Float8, float, float>;
```
Comment (jichangjichang) [inline tensilelite/Tensile/Source/client/include/TypedId.hpp:122]: better to add this config with F8 impl PR

Comment (KKyang) [reply tensilelite/Tensile/Source/client/include/TypedId.hpp]: Fixed

File: tensilelite/Tensile/Common.py (hunk 2 above)
Comment (KKyang) [reply tensilelite/Tensile/Common.py]: Fixed

**Review by jichangjichang — APPROVED** [review]
LGTM

**Pull request merged**